server/src/main/scala/org/scassandra/server/cqlmessages/AbstractMessageFactory.scala

```diff
@@ -86,10 +86,15 @@ abstract class AbstractMessageFactory extends CqlMessageFactory {
     val query = CqlProtocolHelper.readLongString(iterator)
     val consistency = iterator.getShort
     val flags = iterator.getByte
-    val numberOfVariables = iterator.getShort
-    val variableValues = variableTypes.map (varType => {
-      varType.readValue(iterator, protocolVersionImp)
-    } )
+    val variableValues = flags & 0x1 match {
+      case 0x1 => {
+        val numberOfVariables = iterator.getShort
+        variableTypes.map (varType => {
+          varType.readValueWithLength(iterator)(protocolVersionImp)
+        })
+      }
+      case _   => Nil
+    }
```

Comment:
Ah nice, yep it's the same thing I believe.  It doesn't look like QueryFlagParser is used except for Testing, but it would apply perfectly here.  I'll change it to use it.